internal/comby/comby_test.go

```diff
@@ -0,0 +1,139 @@
+package comby
+
+import (
+	"archive/tar"
+	"bufio"
+	"bytes"
+	"context"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"testing"
+
+	"github.com/sourcegraph/sourcegraph/internal/api"
+	"github.com/sourcegraph/sourcegraph/internal/gitserver"
+	"github.com/sourcegraph/sourcegraph/internal/store"
+)
+
+func TestMatchesInZip(t *testing.T) {
+	// If we are not on CI skip the test if comby is not installed.
+	if _, err := exec.LookPath("comby"); os.Getenv("CI") == "" && err != nil {
+		t.Skip("comby is not installed on the PATH. Try running 'bash <(curl -sL get.comby.dev)'.")
+	}
+
+	files := map[string]string{
+
+		"README.md": `# Hello World
+
+Hello world example in go`,
+		"main.go": `package main
+
+import "fmt"
+
+func main() {
+	fmt.Println("Hello foo")
+}
+`,
+	}
+
+	zipPath, cleanup, err := newZip(files)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer cleanup()
+
+	cases := []struct {
+		args Args
+		want string
+	}{
+		{
+			args: Args{
+				Input:           ZipPath(zipPath),
+				MatchTemplate:   "func",
+				RewriteTemplate: "derp",
+				FilePatterns:    []string{".go"},
+				Matcher:         ".go",
+			},
+			want: `{"uri":"main.go","diff":"--- main.go\n+++ main.go\n@@ -2,6 +2,6 @@\n \n import \"fmt\"\n \n-func main() {\n+derp main() {\n \tfmt.Println(\"Hello foo\")\n }"}
+`},
+	}
+
+	for _, test := range cases {
+		b := new(bytes.Buffer)
+		w := bufio.NewWriter(b)
+		err := PipeTo(test.args, w)
+		if err != nil {
+			t.Fatal(err)
+		}
+		got := b.String()
+		if got != test.want {
+			t.Errorf("got %v, want %s", got, test.want)
```

Comment:
Please accept only if you think it's appropriate:
```suggestion
			t.Errorf("got %v, want %v", got, test.want)
```

Refs: https://sourcegraph.com/search?q=repo:%5Egithub%5C.com/sourcegraph/sourcegraph%24+t.Errorf%28%22got+%25v&patternType=literal#10